File: mcdowell/python/ch2-03.py
from ch2_linked_list import LinkedList


def deleteNode(node):
    if node is None or node.next is None:
        return False

    nextNode = node.next
    node.val = nextNode.val
    node.next = nextNode.next

    return True


def test():
    l = LinkedList()
    l.addValue(1)
    l.addValue(2)
    l.addValue(3)
    node = l.addValue(32)
    l.addValue(30)
    l.addValue(3)
    l.addValue(4)
    l.addValue(5)
    node = deleteNode(node)

    print l.getAll()



if __name__ == '__main__':
    test()

File: mcdowell/php/ch2-03.php
<?php
require "ch2-linked-list.php";


function deleteNode(LinkedListNode $node) {
    if ($node == null || $node->next == null) return false;
    $nextNode = $node->next;
    $node->val = $nextNode->val;
    $node->next = $nextNode->next;
    return true;
}



$l = new LinkedList();
$l->addValue(2);
$l->addValue(3);
$l->addValue(5);
$watchNode = $l->addValue(9);
$l->addValue(20);
$l->addValue(5);
$l->addValue(5);
$l->addValue(25);


deleteNode($watchNode);
print_r($l->getAll());